examples/src/main/java/org/apache/druid/examples/twitter/TwitterSpritzerFirehoseFactory.java

```diff
@@ -162,10 +165,12 @@ public void onStatus(Status status)
           throw new RuntimeException("Interrupted, time to stop");
         }
         try {
-          boolean success = queue.offer(status, 15L, TimeUnit.SECONDS);
-          if (!success) {
-            log.warn("queue too slow!");
-          }
+          statusQueueHelper.offerAndHandleFailure(
```

Comment:
In the current form `BlockingQueueHelper` only increases the complexity of classes where it is used. It's too shallow.

I doubt it's reasonable to extract the pattern from all places across the codebase where it was found. What about abstracting it just in emitters? See #7057.

Reply:
Makes sense.  I noticed the behaviour was fairly different between the provided examples so I used this abstraction to avoid changing behaviour.  Does it make sense to close this issue/PR and focus on #7057?

Reply:
I think so.